### PR TITLE
refactor: use ESM import from esm.sh

### PR DESCRIPTION
**Pull request by manzt**
A better implementation of #98.

> Rather than (painfully) dealing with global module systems in various runtimes (e.g., requirejs), this PR uses a scoped ESM import for gosling from esm.sh, an ESM only cdn.

> These URLs aren't as stable as unpkg, but the imports should work more consistently.

### CODE REVIEW

**Comment by manzt** [comment]
I think this will make it easier for users to create their own custom renderers as well.

File: gosling/display.py [hunk 1]
```diff
@@ -13,68 +13,17 @@
...
-            script.onload = resolve;
-            script.src = src;
-            script.async = false;
-            document.head.appendChild(script);
-        });
-    }
-
-    async function loadGosling() {
-        // Manually load scripts from window namespace since requirejs might not be
-        // available in all browser environments.
```
Comment (sehilyi) [inline gosling/display.py:34]: I assume we needed these part since Jupyter Lab/Notebook did not support requirejs and the use of esm.sh addresses this issue?

**Review by sehilyi — APPROVED** [review]
Nice! I can see that resulting HTML file is much more concise without the reuirejs-related code, which I think makes it easier for users to understand the exported code.

Just curious – is there any notable drawback when using `esm.sh`?

**Comment by manzt** [comment]
TL;DR - Notebook environments rely heavily on different types of _global_ module systems. RequireJS is used in Jupyter Notebook, but not Colab or Jupyter Lab. This created a massive headache to try to make a single HTML template that works in all these environments. 

Browsers now support ESM natively. The main benefit is that we can rely on this behavior across environments (since it's built into the browser) rather than inspecting if or if not something like RequireJS is available. ESM allows us to import gosling without polluting some global namespace (relying on `window.gosling` to be available). 

> Just curious – is there any notable drawback when using esm.sh?

This is a great question. I guess the most notable difference is that the esm.sh URL isn't a CDN over static files published to npm (like unpkg is just a mirror for any files pushed to NPM), but rather a bundler + CDN. This means that we are dependent on esbuild being able to bundle `gosling.js`, but I think it's an ok tradeoff.
